### PR TITLE
Fix a few socket test failures on Android

### PR DESCRIPTION
**Pull request by alexrp**
Fixes 3 of the 4 socket failures that we're currently seeing on Android.

### CODE REVIEW

**Comment by alexrp** [comment]
@esdrubal please review

**Comment by esdrubal** [comment]
This is changing the code to not throw an exception. We decided to add an exception to fail fast when double binding is not available on the underlying system, but there are some tests that do double binding and are not handling it, that is why they fail.

We can add the try catch block around those double binding failing tests.
Or we can also reconsider the exception altogether, because well written apps that may already handle double bind exceptions when the bind is done will now fail when the ReuseAddress is set to true.

I think that making the ReuseAddress to fail silently (remain false) may avoid us and user apps unnecessary issues.

**Comment by alexrp** [comment]
@esdrubal How is this changing the code to not throw an exception? All it's doing is making it so that the check happens in the correct socket option "namespace". If you passed something like `SocketOptionLevel.IP` + `SocketOptionName.IpTimeToLive`, the code would throw a `SocketException` on systems that don't support address reuse (which is completely unrelated to the `IpTimeToLive` option) because it misinterprets the `IpTimeToLive` value as `ReuseAddress` (they share the same enumeration value).

If you look throughout `Socket.cs`, all other comparisons of `SocketOptionName` values are accompanied by a comparison of a `SocketOptionLevel` value as well.

**Comment by esdrubal** [comment]
@alexrp thanks for the explanation, I assumed that this was changing the code to not throw an exception because the test change shouldn't be required. It is intended to when ReuseAddress is set not when the bind is done.

**Comment by alexrp** [comment]
@esdrubal so 36fbe26be71eddab369dca9c6ed68cfc8c8ab531 looks good then, right?

I'll have a closer look at 07b201e8a32b086779f5a1838a68e3121b8a34fe.

**Comment by alexrp** [comment]
Rebased after some discussion with @esdrubal.

File: mcs/class/System/Test/System.Net.Sockets/SocketTest.cs [hunk 1]
```diff
@@ -3440,6 +3440,7 @@ public void UdpDoubleBind ()
 
 				try {
 					ss.Bind (new IPEndPoint (IPAddress.Any, ep.Port));
+					ss.Listen (1);
```
Comment (esdrubal) [inline mcs/class/System/Test/System.Net.Sockets/SocketTest.cs]: Was this listen enough to make the test fail? Or do we have to do listen on the first socket too?

**Comment by esdrubal** [comment]
LGTM thanks for taking care of the mess I created here. :+1:

**Comment by alexrp** [comment]
@monojenkins merge

**Comment by alexrp** [comment]
amd64 failure is unrelated, merging.

**Pull request merged**